packages/now-routing-utils/src/superstatic.ts

```diff
@@ -197,13 +186,25 @@ function replaceSegments(
       hash,
     });
 
-    // url.format() escapes the query string but we must preserve dollar signs
-    destination = destination.replace(/=%24/g, '=$');
+    // url.format() escapes the dollar sign but it must be preserved for now-proxy
+    destination = destination.replace(/%24/g, '$');
   }
 
   return destination;
 }
 
+function safelyCompile(str: string, indexes: { [k: string]: string }): string {
+  if (!str) {
+    return str;
+  }
+  // path-to-regexp cannot compile question marks
+  return str
+    .split('?')
```

Comment:
What if the user tries to do an optional directly followed by a query do we need to de-dupe the query for this case e.g. `/hello/:world??another=value`?

Reply:
What output would you expect for this input? I can add another test 😃

Reply:
I think it would be expected for `/hello/:world??another=value` to become `/hello/$1?another=value&world=$1`

Reply:
Hmm, that seems to cause an issue with `url.parse()` and then `url.format()`.

```js
const destination = '/hello/:world??another=value';
const parsedDestination = url.parse(destination, true);
delete parsedDestination.href;
delete parsedDestination.path;
delete parsedDestination.search;
url.format(parsedDestination);
//    /api/hello/:world?%3Fanother=value
```

Reply:
Interesting, it may be a non-issue since it is a weird syntax. We can monitor to see if it comes up